vim/merlin/autoload/merlin.vim

```diff
@@ -376,14 +376,34 @@ endfunction
 
 function! merlin#Locate(...)
   if (a:0 > 1)
-    echoerr "Locate: to many arguments (expected 0 or 1)"
+    echoerr "Locate: too many arguments (expected 0 or 1)"
   elseif (a:0 == 0) || (a:1 == "")
     MerlinPy merlin.vim_locate_under_cursor()
   else
     MerlinPy merlin.vim_locate_at_cursor(vim.eval("a:1"))
   endif
 endfunction
 
+function! merlin#LocateImpl(...)
```

Comment:
Naïve question (because I haven't touched vimscript in years), why can't this function be implemented as:
```vim
let l:pref = g:merlin_locate_preference
let g:merlin_locate_preference = 'ml'
call merlin#Locate(...)
let g:merlin_locate_preference = l:pref
```

Reply:
Updated to do roughly this. Vimscript being vimscript, your elegant call syntax needed to be replaced with something much uglier and less obvious:
```vim
call call("merlin#Locate", a:000)
```